systemtests/src/test/java/enmasse/systemtest/mqtt/MqttTestBase.java

```diff
@@ -45,12 +64,126 @@ public void teardownMqttTest() throws Exception {
         this.clients.clear();
     }
 
-    protected MqttClient createClient() {
+    protected MqttClient createClient() throws Exception {
 
-        Endpoint mqttEndpoint = this.openShift.getEndpoint("mqtt", "mqtt");
+        MqttConnectOptions options = new MqttConnectOptions();
+        Endpoint mqttEndpoint;
 
-        MqttClient client = new MqttClient(mqttEndpoint);
+        if (environment.useTLS()) {
+
+            mqttEndpoint = openShift.getRouteEndpoint("mqtt");
+
+            SSLContext sslContext = tryGetSSLContext("TLSv1.2", "TLSv1.1", "TLS", "TLSv1");
+            sslContext.init(null, new X509TrustManager[]{new X509TrustManager() {
+                @Override
+                public void checkClientTrusted(X509Certificate[] chain,
+                                               String authType) throws CertificateException {
+                }
+
+                @Override
+                public void checkServerTrusted(X509Certificate[] chain,
+                                               String authType) throws CertificateException {
+                }
+
+                @Override
+                public X509Certificate[] getAcceptedIssuers() {
+                    return new X509Certificate[0];
+                }
+            }}, new SecureRandom());
+            SSLSocketFactory sslSocketFactory = new SNISettingSSLSocketFactory(sslContext.getSocketFactory(), mqttEndpoint.getHost());
+
+            options.setSocketFactory(sslSocketFactory);
+
+            if (!TestUtils.resolvable(mqttEndpoint)) {
+                mqttEndpoint = new Endpoint("localhost", 443);
```

Comment:
I thought it odd - but I copied it from AmqpTestBase, line 77, I was aiming for consistency even if it seemed an odd choice :-)